src/theory/quantifiers/sygus/sygus_unif_rl.h

```diff
@@ -300,6 +299,52 @@ class SygusUnifRl : public SygusUnif
      private:
       /** reference to parent unif util */
       DecisionTreeInfo* d_dt;
+      /** true and false nodes */
+      Node d_true;
+      Node d_false;
+      /** cache of conditions evaluations on heads */
+      std::map<std::pair<Node, Node>, Node> d_eval_cond_hd;
+
+      /** rebuild decision tree using information gain heuristic
+       *
+       * In a scenario in which the decision tree potentially contains more
+       * conditions than necessary, it is benefetial to rebuild it in a way that
+       * "better" conditions occurr closer to the top.
+       *
+       * The information gain heuristic selects conditions that lead to a
+       * greater reduction of the Shannon entropy in the set of points
+       */
+      void recomputeSolHeuristically(std::map<Node, Node>& hd_mv);
+      /** recursively select (best) conditions to split poinst
+       *
+       * At each call picks the best condition according to information gain and
+       * splits the set of points accordingly, then recurses on them.
+       *
+       * The base case is a set being fully classified (i.e. all points have the
+       * same value)
+       */
+      void buildDt(std::vector<Node>& pts,
```

Comment:
Is this the right name for this function? It seems that this is specific to the heuristic.

Reply:
Also, explain the arguments of this method in a formal way.